src/dogroups.c

```diff
@@ -535,7 +535,7 @@ SEXP growVector(SEXP x, const R_len_t newlen)
   if (isNull(x)) error(_("growVector passed NULL"));
   PROTECT(newx = allocVector(TYPEOF(x), newlen));   // TO DO: R_realloc(?) here?
   if (newlen < len) len=newlen;   // i.e. shrink
-  switch (TYPEOF(x)) {
+  if (len) switch (TYPEOF(x)) {
```

Comment:
I wonder if it'll be clearer to be more explicit here, WDYT?

```suggestion
  if (!len) { // cannot memcpy invalid pointer, #6819
    keepattr(newx, x)
    UNPROTECT(1);
    return newx;
  }
  switch (TYPEOF(x)) {
```